app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js

```diff
@@ -9,18 +9,26 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.$module.hideCookieMessage = this.hideCookieMessage.bind(this)
     this.$module.showConfirmationMessage = this.showConfirmationMessage.bind(this)
     this.$module.setCookieConsent = this.setCookieConsent.bind(this)
+    this.$module.currentPage = window.location.pathname
 
-    this.$hideLink = this.$module.querySelector('a[data-hide-cookie-banner], button[data-hide-cookie-banner]')
-    if (this.$hideLink) {
-      this.$hideLink.addEventListener('click', this.$module.hideCookieMessage)
-    }
+    var newCookieBanner = document.querySelector('.gem-c-cookie-banner--new')
 
-    this.$acceptCookiesLink = this.$module.querySelector('button[data-accept-cookies]')
-    if (this.$acceptCookiesLink) {
-      this.$acceptCookiesLink.addEventListener('click', this.$module.setCookieConsent)
-    }
+    // Hide the cookie banner on the cookie settings page, to avoid circular journeys
+    if (newCookieBanner && this.$module.currentPage === '/help/cookies') {
+      this.$module.style.display = 'none'
+    } else {
+      this.$hideLink = this.$module.querySelector('a[data-hide-cookie-banner], button[data-hide-cookie-banner]')
```

Comment:
`querySelector` will only return one element,  but the selector used looks like it's trying to grab multiple things. Does this need to be a `querySelectorAll`?

Perhaps even a `this.$module.find('a[data-hide-cookie-banner], button[data-hide-cookie-banner]')`, which would return multiple jQuery objects we can `.on('click')`.

Reply:
@ChrisBAshton This is only trying to select multiple things because we have 2 variations of the banner now (current & new). I'd only expect one to ever be present at any one time on a page (and this code will be removed once we switch to the new banner).

Reply:
Ah, great 👍 thanks for explaining!